c2rust-analyze/src/main.rs

```diff
@@ -73,40 +163,85 @@ fn run(tcx: TyCtxt) {
             equiv.unify(a, b);
         }
 
-        func_info.push((acx.into_data(), dataflow, l_equiv));
+        let mut info = FuncInfo::default();
+        info.acx_data.set(acx.into_data());
+        info.dataflow.set(dataflow);
+        info.l_equiv.set(l_equiv);
+        func_info.insert(ldid, info);
     }
 
-    // Compute permission and flag assignments.
-
+    // Remap pointers based on equivalence classes, so all members of an equivalence class now use
+    // the same `PointerId`.
     let (g_counter, g_equiv_map) = g_equiv.renumber();
     eprintln!("g_equiv_map = {:?}", g_equiv_map);
     gacx.remap_pointers(&g_equiv_map, g_counter);
-    let mut gasn = GlobalAssignment::new(0, PermissionSet::UNIQUE, FlagSet::empty());
-    for (ldid, info) in tcx.hir().body_owners().zip(func_info.into_iter()) {
-        let ldid_const = WithOptConstParam::unknown(ldid);
-        let name = tcx.item_name(ldid.to_def_id());
-        let mir = tcx.mir_built(ldid_const);
-        let mir = mir.borrow();
 
-        let (mut data, mut dataflow, l_equiv) = info;
-        // Remap pointers based on equivalence classes, so all members of an equivalence class now
-        // use the same `PointerId`.
-        let (l_counter, l_equiv_map) = l_equiv.renumber(&g_equiv_map);
+    for ldid in tcx.hir().body_owners() {
+        let info = func_info.get_mut(&ldid).unwrap();
+        let (l_counter, l_equiv_map) = info.l_equiv.renumber(&g_equiv_map);
```

Comment:
Why is `l` sometimes used without a separator (like `l*`) and sometimes with a `_` (like `l_*`)?

Reply:
`ldid` is (roughly) an initialism for `LocalDefId`, similar to `did` for `DefId`.  The latter appears pervasively within rustc.